docs/proposals/diamond-design.md

```diff
@@ -0,0 +1,458 @@
+```
+my-conflicting-file.data-0ujssxh0cECutqzMgbtXSGnjorm
+```
+
+This creates a new bundle with the specified conflicts resolved (the `-{split-id}` suffix is trimmed from the resulting file).
+Resolved conflicts are removed from the conflicts list in the new bundle. When there is no remaining conflict, the conflicts metadata are wiped out.
+
+
+If everything is fine and you just want to make it clear for all consumers of this bundle, you may force all conflicts to be resolved at once.
+```
+datamon bundle conflicts resolve --bundle {bundle-id} --resolve-all [--message {...} --label {...}]
+```
+This creates a new bundle with the same data (no upload here) and conflicts metadata are wiped out.
+The bundle on which conflicts where initially detected remains untouched for traceability.
+
+### Cancel a diamond
+```
+datamon bundle diamond cancel --repo {repo} --diamond {diamond-id}
+```
+
+This waits for ongoing running splits to complete then marks vmetadata ready for cleanup
+
+**NOTE**: in the future, we may interrupt running splits rather than wait for them to complete _then_ cancel the dimaond
+
+### Identify splits with some custom node identifier
+```
+datamon bundle diamond split add --repo {repo} --path {source} --split-tag {my-bespoke-distinctive-tag}
+```
+
+Tags are kept in metadata and can be used when reporting about running splits or help latter with conflict resolution.
+Tags are typically a hostname or something that helps the end-user to identify a node in the diamond.
+
+## Nice to have / future
+
+   1. `add` commands can be strongly ordered based on a design similar to WAL (KSUIDs alone do not guarantee strong ordering).
+      This is a nice to have addition and can be added later in the implementation.
+   2. explicit cancellation could attempt to interrupt ongoing splits
+
+## Recap of new commands or new options to existing commands
+
+```
+datamon bundle diamond
+              |_  initialize --repo {repo}
+              |
+              |_  commit --repo {repo} --diamond {diamond-id} --message {commit message} [--no-conflict] [--label]
+              |
+              |_  list --repo {repo}
+              |
+              |_  cancel --repo {repo} --diamond {diamond-id}
+              |
+              |_  split
+                    |
+                    |_ add --repo {repo} --diamond {diamond-id} --path {source} [--split-tag {my-bespoke-distinctive-tag}] [--files {list of files}]
+                    |
+                    |_ list --repo {repo} --diamond {diamond-id}
+
+
+datamon bundle conflicts resolve --bundle {bundle-id} {--files {file-list}|--resolve-all} [--message {...} --label {...}]
+
+datamon bundle get --repo {repo} --bundle {bundle-id} [--with-conflicts]
+
+datamon bundle get --repo {repo} --bundle {bundle-id} [--with-diamonds]
+
+datamon bundle list files --repo {repo} --bundle {bundle-id} [--with-conflicts|--with-checkpoints]
```

Comment:
drop. Once a bundle it should be a normal bundle. I do not want to introduce types of bundles.

Reply:
Was intended for audit trail inspection not typing bumdles

Reply:
ok dropped.
Notice that in the future, inspecting / reporting about diamonds & splits will cause the same kind of headache as we have now to join labels with bundles (typically the join operation is delegated to some - slower - porcelain func).